src/common/abstractPromote.ts

```diff
@@ -75,7 +76,11 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
       this.flags['branch-name']
     );
     this.sourceStageId = this.getSourceStageId();
-    const asyncOperationId: string = await this.requestPromotion(doceOrg);
+    const asyncOperationId: string = await this.retryEnabledRequestPromotion(doceOrg, this.numRetries409);
+    this.spinner.stop();
+    if (asyncOperationId?.includes('errorCode')) {
+      throw new Error(JSON.stringify(asyncOperationId)); // Is this the type of error we want to throw here?
```

Comment:
If there's a specific type of other error we want to construct and throw, then here's the place to do it. otherwise I'll delete my comment here

Reply:
Yeah, this was something I talked over with Esteban just now and I changed how I'm doing this so it's handled better.